cpp/tests/unary/unary_ops_test.cu

```diff
@@ -709,3 +709,104 @@ TYPED_TEST(CastFromTimestamps, WithNulls) {
+
+  std::unique_ptr<cudf::column> got = cudf::experimental::is_nan(col);
+
+  cudf::test::expect_columns_equal(expected, got->view());
+}
+
+TYPED_TEST(IsNAN, WithNull) {
+  using T = TypeParam;
+
+  // The last NAN is null
+  cudf::test::fixed_width_column_wrapper<T> col {{1, 2, NAN, 4, NAN, 6, 7}, {1, 0, 1, 1, 0, 1, 1}};
+  cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected = {false, false, true, false, false, false, false};
+
+  std::unique_ptr<cudf::column> got = cudf::experimental::is_nan(col);
+
+  cudf::test::expect_columns_equal(expected, got->view());
+}
+
+TYPED_TEST(IsNAN, EmptyColumn) {
+  using T = TypeParam;
+
+  cudf::test::fixed_width_column_wrapper<T> col {};
+  cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected = {};
+
+  std::unique_ptr<cudf::column> got = cudf::experimental::is_nan(col);
+
+  cudf::test::expect_columns_equal(expected, got->view());
+}
+
+TYPED_TEST(IsNAN, NonFloatingColumn) {
+  using T = TypeParam;
+
+  cudf::test::fixed_width_column_wrapper<int32_t> col {{1, 2, 5, 3, 5, 6, 7}, {1, 0, 1, 1, 0, 1, 1}};
+
+  EXPECT_THROW(std::unique_ptr<cudf::column> got = cudf::experimental::is_nan(col), cudf::logic_error);
+}
+
+template <typename T>
+struct IsNotNAN : public cudf::test::BaseFixture {};
+
+using test_types =
+    ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(IsNotNAN, test_types);
+
+TYPED_TEST(IsNotNAN, AllValid) {
+  using T = TypeParam;
+
+  cudf::test::fixed_width_column_wrapper<T> col {{1, 2, NAN, 4, NAN, 6, 7}};
+  cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected = {true, true, false, true, false, true, true};
+
+  std::unique_ptr<cudf::column> got = cudf::experimental::is_notnan(col);
+
+  cudf::test::expect_columns_equal(expected, got->view());
+}
+
+TYPED_TEST(IsNotNAN, WithNull) {
+  using T = TypeParam;
+
+  // The last NAN is null
+  cudf::test::fixed_width_column_wrapper<T> col {{1, 2, NAN, 4, NAN, 6, 7}, {1, 0, 1, 1, 0, 1, 1}};
+  cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected = {true, true, false, true, true, true, true};
+
+  std::unique_ptr<cudf::column> got = cudf::experimental::is_notnan(col);
+
+  cudf::test::expect_columns_equal(expected, got->view());
+}
+
+TYPED_TEST(IsNotNAN, EmptyColumn) {
+  using T = TypeParam;
+
+  cudf::test::fixed_width_column_wrapper<T> col {};
+  cudf::test::fixed_width_column_wrapper<cudf::experimental::bool8> expected = {};
+
+  std::unique_ptr<cudf::column> got = cudf::experimental::is_notnan(col);
+
+  cudf::test::expect_columns_equal(expected, got->view());
+}
+
+TYPED_TEST(IsNotNAN, NonFloatingColumn) {
+  using T = TypeParam;
+
+  cudf::test::fixed_width_column_wrapper<int64_t> col {{1, 2, 5, 3, 5, 6, 7}, {1, 0, 1, 1, 0, 1, 1}};
+
+  EXPECT_THROW(std::unique_ptr<cudf::column> got = cudf::experimental::is_notnan(col), cudf::logic_error);
+}
+
```

Comment:
I don't see any test cases for null values.  I want to be sure that we get this right, despite me not having documented in properly in the request.  From looking at the code I think you did it right but I want to be sure that it is tested.

is_nan(null) returns false
Spark does not have an is_notnan, but I would expect it to be the same as not(is_nan(null)) which would be true.

Like I said I think you got it right in the implementation.  I just want to make sure that there are tests for it.

Reply:
- If the element is NULL, then it will false in output column, only valid element with NAN will be true.
- I just created is_notnan if in any case you might require.

Reply:
@revans2 Check the comment below for the test case with null values. One of the NULL values is an NAN and it will result as `false` than `true`.

Reply:
https://github.com/rapidsai/cudf/blob/38309cbc982bbcec3a0819dbebdf61ea87daa00b/cpp/tests/unary/unary_ops_test.cu#L732